hw/templates/snax_accelerator_top_wrapper.sv.tpl

```diff
@@ -0,0 +1,275 @@
+  output logic [TCDMReqPorts-1:0][3:0]                   tcdm_req_amo_o, //Note that tcdm_req_amo_i is 4 bits based on reqrsp definition
+  output logic [TCDMReqPorts-1:0][NarrowDataWidth-1:0]   tcdm_req_data_o,
+  output logic [TCDMReqPorts-1:0][4:0]                   tcdm_req_user_core_id_o, //Note that tcdm_req_user_core_id_i is 5 bits based on Snitch definition
+  output logic [TCDMReqPorts-1:0]                        tcdm_req_user_is_core_o,
+  output logic [TCDMReqPorts-1:0][NarrowDataWidth/8-1:0] tcdm_req_strb_o,
+  output logic [TCDMReqPorts-1:0]                        tcdm_req_q_valid_o,
+
+  // Response
+  input  logic [TCDMReqPorts-1:0]                        tcdm_rsp_q_ready_i,
+  input  logic [TCDMReqPorts-1:0]                        tcdm_rsp_p_valid_i,
+  input  logic [TCDMReqPorts-1:0][NarrowDataWidth-1:0]   tcdm_rsp_data_i,
+
+  //-----------------------------
+  // CSR control ports
+  //-----------------------------
+  // Request
+  input  logic [31:0] snax_req_data_i,
+  input  logic [31:0] snax_req_addr_i,
+  input  logic        snax_req_write_i,
+  input  logic        snax_req_valid_i,
+  output logic        snax_req_ready_o,
+
+  // Response
+  input  logic        snax_rsp_ready_i,
+  output logic        snax_rsp_valid_o,
+  output logic [31:0] snax_rsp_data_o
+);
+
+  //-----------------------------
+  // Internal local parameters
+  //-----------------------------
+  localparam int unsigned NumCsr = ${cfg["snax_acc_num_csr"]};
+
+  //-----------------------------
+  // Accelerator ports
+  //-----------------------------
+  // Input ports from accelerator to streamer
+  logic [ NumInputPorts-1:0][StreamerDataWidth-1:0] acc2stream_data;
+  logic [ NumInputPorts-1:0]                        acc2stream_valid;
+  logic [ NumInputPorts-1:0]                        acc2stream_ready,
+
+  // Ouput ports from accelerator to streamer
+  logic [NumOutputPorts-1:0][StreamerDataWidth-1:0] stream2acc_data;
+  logic [NumOutputPorts-1:0]                        stream2acc_valid;
+  logic [NumOutputPorts-1:0]                        stream2acc_ready;
+
+  // CSR MUXing
+  logic [1:0][RegAddrWidth-1:0] acc_csr_req_addr;
+  logic [1:0][RegDataWidth-1:0] acc_csr_req_data;
+	logic [1:0]                   acc_csr_req_wen;
+	logic [1:0]                   acc_csr_req_valid;
+	logic [1:0]                   acc_csr_req_ready;
+	logic [1:0][RegDataWidth-1:0] acc_csr_rsp_data;
+	logic [1:0]                   acc_csr_rsp_valid;
+	logic [1:0]                   acc_csr_rsp_ready;
+
+  // Register set signals
+  logic [NumCsr-1:0][31:0] acc_csr_reg_set;
+  logic                    acc_csr_reg_set_valid;
+  logic                    acc_csr_reg_set_ready;
+
+  //-------------------------------
+  // MUX and DEMUX for control signals
+  // That separate between streamer CSR
+  // and accelerator CRS
+  //-------------------------------
+  csr_mux_demux #(
+    .AddrSelOffSet        ( 8                     ),
```

Comment:
Why the CSR address offset is always 8? What if the accelerator has more than 8 CSRs?